Ci

macpijan opened the pull request.

macpijan committed:
.github/workflows/build.yml: add

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
pre-commit: add hooks

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
pre-commit: autofixes

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
pre-commit: replace shellcheck provider

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
pre-commit: skip rustfmt on CI

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
pre-commit: skip shellcheck on CI

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

macpijan committed:
README.md: remove shellcheck as a dependency

Signed-off-by: Maciej Pijanowski <maciej.pijanowski@3mdeb.com>

arturkow2000 commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,34 @@
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
`actions/checkout` has support for cloning submodules
```suggestion
        uses: actions/checkout@v2
        submodules: recursive
```

macpijan replied:
Please try that and if it works we can use that. I copied it over from another project, where there were some isues with the builtin submodule clone.

arturkow2000 replied:
Done

arturkow2000 commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,34 @@
+  push:
+
+env:
+  FOBNAIL_SDK_VERSION: v0.2.3
```
@macpijan Updating to v0.2.4 (after https://github.com/fobnail/fobnail-sdk/pull/8) should fix problem with building on nRF.
```suggestion
  FOBNAIL_SDK_VERSION: v0.2.4
```

macpijan replied:
please change and test

arturkow2000 replied:
Done

arturkow2000 committed:
Fix building on nRF target.

Update to SDK v0.2.4 which has fixes necessary for successful build.
Enable optimizations so that compiled binary fits into 1 MiB flash.

arturkow2000 commented:
@macpijan I updated to latest SDK, rebased this PR to main and enabled optimizations so that the latest code still builds (binary size crossed 1 MiB).
https://github.com/fobnail/fobnail/runs/5478684759?check_suite_focus=true

tomzy-0 commented:
@arturkow2000 when you start using pre-commit in any repo please run local check, now we got 1 commit from the bot - could you please check it? Also the typo hook needs to be checked, now it fails on

```
typos....................................................................Failed
- hook id: typos
- exit code: 2
- files were modified by this hook

error: `PARMS` should be `PARAMS`, `PRAMS`
  --> src/client/tpm/mu.rs:245:30
    |
245 |         // Decode TPMU_PUBLIC_PARMS, contents depend on algorithm type
    |                               ^^^^^
    |
```
But I believe that the name is correct here. Maybe we can add some exceptions to this hook?

Commit:
src: fix typos after running typo pre-commit hook

Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

Commit:
tools: fix typos after running typo pre-commit hook

Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

Commit:
typos.toml: add config for typos pre-commit hook

Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

tomzy-0 commented:
@arturkow2000 I added some configs for typo hooks. Could you please verify if everything looks OK in this PR?

Commit:
.pre-commit-config.yaml: bump typos hook to v1.5.0

Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

Commit:
typos.toml: remove extend-exclude option

Signed-off-by: Tomasz Żyjewski <tomasz.zyjewski@3mdeb.com>

krystian-hebel commented on Cargo.toml:
```diff
@@ -43,26 +43,8 @@ aes = "0.8"
-p256-cortex-m4 = { opt-level = "z" }
-salty = { opt-level = "z" }
-serde-indexed = { opt-level = "z" }
+# FIXME: Littlefs requires strcpy() when optimizations are one.
```
Not sure if this was the intention
```suggestion
# FIXME: Littlefs requires strcpy() when optimizations are done.
```

arturkow2000 replied:
Fixed, I meant that it requires strcpy when optimizations are enabled.

krystian-hebel commented on README.md:
````diff
@@ -0,0 +1,29 @@
+
+```
+pre-commit install
+```
````
This puts significant software requirements on users (Python and Rust). Since we're using docker anyway for building, would it be possible to put those inside `fobnail-sdk` and start a container from `pre-commit` hook?

arturkow2000 replied:
@krystian-hebel Got working pre-commit. Requires latest SDK https://github.com/fobnail/fobnail-sdk/pull/9
See README.md for instructions.

arturkow2000 committed:
Cargo.toml: fix unclear statement

arturkow2000 commented:
@TomaszAIR
TPMU_PUBLIC_PARMS is correct.
~~I added one more exception to typos.toml - typos was correcting "choosed" word on `rims` branch.~~
@krystian-hebel Ok, I removed this.

macpijan commented:
`TPMU_PUBLIC_PARMS` comes from some external library or defined by us? `TPMU_PUBLIC_PARAMS` sounds more sane to me as well.

... unless it means something else than `parameters` ...

arturkow2000 committed:
Use pre-commit from Fobnail SDK

arturkow2000 commented:
@macpijan `TPMU_PUBLIC_PARMS` comes from libtss2

krystian-hebel commented:
And originally from TPM library specification: https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf

arturkow2000 committed:
Support Shellcheck in Docker. Use SDK v0.2.5

arturkow2000 committed:
README.md: update information about pre-commit hooks

krystian-hebel commented:
I can't pinpoint exact issue, but `./pre-commit.sh run` fails:

```
$ ./pre-commit.sh run
useradd: warning: the home directory already exists.
Not copying any file from skel directory into it.
An unexpected error has occurred: PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 70, in error_handler
    yield
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/main.py", line 353, in main
    store = Store()
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 61, in __init__
    with self.exclusive_lock():
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 89, in exclusive_lock
    with file_lock.lock(os.path.join(self.directory, '.lock'), blocked_cb):
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/file_lock.py", line 74, in lock
    with open(path, 'a+') as f:
PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/pre-commit", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/main.py", line 409, in main
    f'Command {args.command} failed to exit with a returncode',
  File "/usr/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 78, in error_handler
    _log_and_exit(msg, ret_code, e, traceback.format_exc())
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 28, in _log_and_exit
    storedir = Store().directory
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 61, in __init__
    with self.exclusive_lock():
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 89, in exclusive_lock
    with file_lock.lock(os.path.join(self.directory, '.lock'), blocked_cb):
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/file_lock.py", line 74, in lock
    with open(path, 'a+') as f:
PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'
```

Seems that `pre-commit` directory has wrong owner when created by https://github.com/fobnail/fobnail/pull/16/files#diff-08c4d6d6a814d6049c5bbe4df3f827b6229276467a22774f0f22047d5d2eebd9R20

```
$ ls -l .temp/
drwxrwxr-x 4 krystian krystian 4096 lut 28 15:03 cargo
drwxr-xr-x 2 root     root     4096 mar 15 19:44 pre-commit
```

This does not happen with `git commit`, assuming that directory was manually removed, but ShellCheck fails in that case. This may be leftover from previous installation without docker, I haven't checked on clear installation.

krystian-hebel commented:
I can't pinpoint exact issue, but `./pre-commit.sh run` fails:

```
$ ./pre-commit.sh run
useradd: warning: the home directory already exists.
Not copying any file from skel directory into it.
An unexpected error has occurred: PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 70, in error_handler
    yield
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/main.py", line 353, in main
    store = Store()
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 61, in __init__
    with self.exclusive_lock():
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 89, in exclusive_lock
    with file_lock.lock(os.path.join(self.directory, '.lock'), blocked_cb):
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/file_lock.py", line 74, in lock
    with open(path, 'a+') as f:
PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/pre-commit", line 10, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/main.py", line 409, in main
    f'Command {args.command} failed to exit with a returncode',
  File "/usr/lib/python3.7/contextlib.py", line 130, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 78, in error_handler
    _log_and_exit(msg, ret_code, e, traceback.format_exc())
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/error_handler.py", line 28, in _log_and_exit
    storedir = Store().directory
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 61, in __init__
    with self.exclusive_lock():
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/store.py", line 89, in exclusive_lock
    with file_lock.lock(os.path.join(self.directory, '.lock'), blocked_cb):
  File "/usr/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.7/dist-packages/pre_commit/file_lock.py", line 74, in lock
    with open(path, 'a+') as f:
PermissionError: [Errno 13] Permission denied: '/home/builder/.cache/pre-commit/.lock'
```

Seems that `pre-commit` directory has wrong owner when created by https://github.com/fobnail/fobnail/pull/16/files#diff-08c4d6d6a814d6049c5bbe4df3f827b6229276467a22774f0f22047d5d2eebd9R20

```
$ ls -l .temp/
drwxrwxr-x 4 krystian krystian 4096 lut 28 15:03 cargo
drwxr-xr-x 2 root     root     4096 mar 15 19:44 pre-commit
```

This does not happen with `git commit`, assuming that directory was manually removed, but ShellCheck fails in that case. This may be leftover from previous installation without docker, I haven't checked on clear installation.

arturkow2000 committed:
pre-commit.sh: fix permission problems, pass docker socket from hook

Manually create .temp/pre-commit before spawning container, previously
this directory was created only when calling pre-commit from git hook
leading to permission problems if `pre-commit.sh run` was called before.
Pass host's Docker socket not only from pre-commit.sh but also from
installed hook to fix Shellcheck.

arturkow2000 commented:
@krystian-hebel If `.temp/pre-commit` doesn't exist then Docker creates it giving root:root owner. Fixed by creating it manually before calling container.

macpijan commented:
@krystian-hebel @arturkow2000 If shellcheck is the issue, let's give up on this hook. We do not plan to ship much more scripts here after all...

macpijan commented:
Also, if we want to run the hooks in docker, we may also install `shellcheck` in the container we use and use this hook `https://github.com/jumanjihouse/pre-commit-hooks` (it does not use docker, it uses natively installed shellcheck program. This way, there would be no need to call sibling container from one container.

macpijan commented:
> And originally from TPM library specification: https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf

It's funny as some results of `TPMU_PUBLIC_PARAMS` can also be found in their docs: https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf

tomzy-0 commented:
@arturkow2000 could you check if installing `shellcheck` in the container and using `pre-commit-hooks` works better?

arturkow2000 committed:
pre-commit.sh: use shellcheck from Fobnail SDK

arturkow2000 commented:
@TomaszAIR @macpijan @krystian-hebel
Shellcheck works properly, however it's old version (v0.5.0) and doesn't support `--severity=warning`.

arturkow2000 commented:
@TomaszAIR
Updated Shellcheck to 0.8.0 (now part of Fobnail SDK) so we don't have to spawn sibling container anymore.
Requires this PR https://github.com/fobnail/fobnail-sdk/pull/9
